YAKL_atomics.h

```diff
@@ -102,9 +102,91 @@
         ::atomicMax( &update , value );
       }
     #endif
-  #endif
 
-  #ifdef __USE_HIP__
+  #elif defined(__USE_SYCL__)
+    template <typename T, sycl::access::address_space addressSpace =
+	      sycl::access::address_space::global_space>
```

Comment:
This is essentially `enable_if` esque, and it will restrict to global space (e.g. not CUDA Shared). YAKL doesn't support Shared memory anyway because it isn't portable.